### PR TITLE
[-Wsign-compare] warning fixes batch 15

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac

### CODE REVIEW

**Review by alextp — DISMISSED** [review]

File: tensorflow/compiler/tf2xla/const_analysis.cc [hunk 1]
```diff
@@ -74,8 +74,8 @@ Status CondConstInputIndices(
         *(fbody->graph), &compile_time_const_arg_indices,
         /*compile_time_const_nodes=*/nullptr, flib_runtime));
   }
-  for (int i = 0, iter_limit = compile_time_const_arg_indices.size();
-       i < iter_limit; i++) {
+  for (int i = 0, end = compile_time_const_arg_indices.size();
+       i < end; i++) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/const_analysis.cc:78]: Should merge in a single line

File: tensorflow/compiler/xla/shape.cc [hunk 2]
```diff
@@ -102,7 +102,8 @@ void Shape::DeleteDimension(int64 dim_to_delete) {
   dynamic_dimensions_.erase(dynamic_dimensions_.begin() + dim_to_delete);
   if (LayoutUtil::HasLayout(*this)) {
     layout_.set_format(DENSE);
-    for (int64 i = 0; i < layout_.minor_to_major().size();) {
+    for (int64 i = 0, end = layout_.minor_to_major().size();
+         i < end;) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/shape.cc:106]: Should merge in a single line

File: tensorflow/compiler/xla/shape_util.cc [hunk 3]
```diff
@@ -524,7 +524,8 @@ ShapeUtil::MakeShapeWithDescendingLayoutAndSamePhysicalLayout(
   }
   string result = StrCat(
       primitive_util::LowercasePrimitiveTypeName(shape.element_type()), "[");
-  for (int i = 0; i < shape.dimensions().size(); i++) {
+  for (int i = 0, end = shape.dimensions().size();
+       i < end; i++) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/shape_util.cc:528]: Should merge in a single line

File: tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc [hunk 4]
```diff
@@ -176,7 +176,8 @@ class GrpcRemoteMaster : public MasterInterface {
               ? deadline_with_backoff_micros
               : expired_time_micros;
       Env::Default()->SleepForMicroseconds(backoff_until - now_micros);
-      if (Env::Default()->NowMicros() > expired_time_micros &&
+      const int64 default_NowMicros = Env::Default()->NowMicros();
+      if (default_NowMicros > expired_time_micros &&
           timeout_in_ms > 0) {
```
Comment (mihaimaruseac) [inline tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:181]: Should merge in a single line

Comment (mihaimaruseac) [reply tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:181]: Also, please use `snake_case` for local variable names

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 5]
```diff
@@ -1567,7 +1567,7 @@ Status StridedSliceTransposer::PermuteMask(TransposeContext* context,
     return errors::InvalidArgument("invalid mask value: ", mask_i);
   }
   int result = 0;
-  for (int i = 0, iter_limit = context->src_to_dst.size(); i < iter_limit;
+  for (int i = 0, end = context->src_to_dst.size(); i < end;
        i++) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1571]: Should merge in a single line

File: tensorflow/core/profiler/internal/tfprof_code.cc [hunk 6]
```diff
@@ -446,12 +447,13 @@ void TFCode::Build() {
...
     CodeNode* leaf = nullptr;
     CodeNode* pre_code_node = root_.get();
-    for (int i = 0; i < fn->call_stack()->traces().size(); ++i) {
+    for (int i = 0, end = fn->call_stack()->traces().size(); i < end; ++i) {
       const string& trace =
           GetTraceString(fn->call_stack()->traces().at(i)) + kGradientSuffix;
       pre_code_node = pre_code_node->AddChildren(
           trace, &fn->call_stack()->traces().at(i), kGradientSuffix);
-      if (i == fn->call_stack()->traces().size() - 1) {
+      const int64 fn_call_stack_traces_size_minus_one = fn->call_stack()->traces().size() - 1;
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/internal/tfprof_code.cc:455]: Let's choose a better name. How about `max_stack_trace_allowed_size`?

File: tensorflow/core/profiler/internal/tfprof_op.h [hunk 7]
```diff
@@ -57,7 +57,8 @@ class TFOp : public TFMultiShow {
 
   bool ShouldShowIfExtra(const ShowMultiNode* node, const Options& opts,
                          int depth) const override {
-    if (opts.min_occurrence > node->node->graph_nodes().size()) {
+    const int node_node_graph_nodes_size = node->node->graph_nodes().size();
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/internal/tfprof_op.h:60]: Let's just use `max_size`/`max_num_graph_nodes`, this name is too long and not that readable

File: tensorflow/core/profiler/utils/derived_timeline.cc [hunk 8]
```diff
@@ -130,7 +130,7 @@ void DerivedXLineBuilder::ExpandOrAddLevelEvent(const XEvent& event,
 }
 
 void DerivedXLineBuilder::ResetLastEvents(int level) {
-  for (int i = level, iter_limit = last_event_by_level_.size(); i < iter_limit;
+  for (int i = level, end = last_event_by_level_.size(); i < end;
        ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/utils/derived_timeline.cc:134]: Should merge in a single line

File: tensorflow/core/profiler/utils/event_span.cc [hunk 9]
```diff
@@ -128,7 +128,7 @@ std::vector<EventTypeSpan> ToNonOverlappedEvents(
   if (event_boundaries.empty()) return result;
   result.reserve(event_boundaries.size());
   PriorityTracker priority_tracker;
-  for (int64 i = 0, iter_limit = (event_boundaries.size() - 1); i < iter_limit;
+  for (int64 i = 0, end = (event_boundaries.size() - 1); i < end;
        i++) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/utils/event_span.cc:132]: Should merge in a single line

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h [hunk 10]
```diff
@@ -39,7 +39,8 @@ class OperandMapping {
   // Given a TFLite index return the ANN index. If it doesn't exist
   // return -1.
   int lite_index_to_ann(int index) const {
-    if (index >= 0 && index < lite_tensor_to_ann_tensor_.size())
+    const int64 lite_tensor_to_ann_tensor_size = lite_tensor_to_ann_tensor_.size();
```
Comment (mihaimaruseac) [inline tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:42]: `max_size` is a much better name. Also in the rest of the change here

File: tensorflow/lite/tools/verifier.cc [hunk 11]
```diff
@@ -106,9 +106,9 @@ bool VerifyStringTensorBuffer(const Tensor& tensor, const Buffer& buffer,
     return false;
   }
   offset += sizeof(int32_t);
-  for (int i = 1; i <= num_strings; i++, offset += sizeof(int32_t)) {
+  for (int i = 1; i <= static_cast<int>(num_strings); i++, offset += sizeof(int32_t)) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/tools/verifier.cc:109]: This is inconsistent with the other rewrites of the loop header throughout these PRs

File: tensorflow/python/client/session_ref.cc [hunk 12]
```diff
@@ -146,7 +146,7 @@ class SessionLogger {
     // Build an index from fetch tensor name to first index in
     // output_tensor_names.
     std::unordered_map<string, int> output_name_to_offset;
-    for (int i = 0, iter_limit = output_tensor_names.size(); i < iter_limit;
+    for (int i = 0, end = output_tensor_names.size(); i < end;
          ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/python/client/session_ref.cc:150]: Should merge in a single line

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/client/session_ref.cc (hunk 12 above)
Comment (tg-at-google) [reply tensorflow/python/client/session_ref.cc:150]: handled in previous PR.

File: tensorflow/core/profiler/utils/event_span.cc (hunk 9 above)
Comment (tg-at-google) [reply tensorflow/core/profiler/utils/event_span.cc:132]: handled in previous PR.

File: tensorflow/core/profiler/utils/derived_timeline.cc (hunk 8 above)
Comment (tg-at-google) [reply tensorflow/core/profiler/utils/derived_timeline.cc:134]: handled in previous PR.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc (hunk 5 above)
Comment (tg-at-google) [reply tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1571]: handled in previous PR.

File: tensorflow/compiler/tf2xla/const_analysis.cc (hunk 1 above)
Comment (tg-at-google) [reply tensorflow/compiler/tf2xla/const_analysis.cc:78]: handled in previous PR.

**Comment by tg-at-google** [comment]
@mihaimaruseac

File: tensorflow/compiler/tf2xla/const_analysis.cc (hunk 1 above)
Comment (mihaimaruseac) [reply tensorflow/compiler/tf2xla/const_analysis.cc:78]: Keeping open until fully handled, the previous PR lands

File: tensorflow/compiler/xla/index_util.cc [hunk 13]
```diff
@@ -157,7 +158,8 @@ namespace xla {
 /* static */ int IndexUtil::CompareIndices(absl::Span<const int64> lhs,
                                            absl::Span<const int64> rhs) {
   int64 rank = lhs.size();
-  CHECK_EQ(rhs.size(), rank);
+  const int64 rhs_size = rhs.size();
+  CHECK_EQ(rhs_size, rank);
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/index_util.cc:162]: Maybe `rhs_rank`?

File: tensorflow/core/profiler/internal/tfprof_code.cc [hunk 14]
```diff
@@ -421,14 +421,15 @@ void TFCode::AddNode(TFGraphNode* node) {
...
+  for (int i = 0, end = node->call_stack()->traces().size(); i < end; ++i) {
     // Unlike op name, which is globally unique, trace name is only unique
     // w.r.t. it's parent.
     const string& trace = GetTraceString(node->call_stack()->traces().at(i));
     traces.insert(trace);
     pre_code_node = pre_code_node->AddChildren(
         trace, &node->call_stack()->traces().at(i), "");
-    if (i == node->call_stack()->traces().size() - 1) {
+    const int64 node_call_stack_traces_size_minus_one = node->call_stack()->traces().size() - 1;
+    if (i == node_call_stack_traces_size_minus_one) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/internal/tfprof_code.cc:432]: `last_index` is much more readable. Consider what happens on a refactor where `call_stack()` gets changed to return an object which now has a `num_traces()` method so you don't need to do `->traces().size()` (a change which can be done automatically by tooling). Then the proposed name will no longer be valid.

File: tensorflow/core/profiler/utils/event_span.cc (hunk 9 above)
Comment (mihaimaruseac) [reply tensorflow/core/profiler/utils/event_span.cc:132]: Keeping unsolved until that lands

Comment (mihaimaruseac) [reply tensorflow/core/profiler/utils/event_span.cc:132]: Keepig unsolved until that lands

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h [hunk 15]
```diff
@@ -60,7 +61,8 @@ class OperandMapping {
 
   // Add a new mapping from `tflite_index` and return the NN API tensor index.
   int add_new_ann_tensor_index(int tflite_index) {
-    if (tflite_index >= lite_tensor_to_ann_tensor_.size()) {
+    const int64 lite_tensor_to_ann_tensor_size = lite_tensor_to_ann_tensor_.size();
+    if (tflite_index >= lite_tensor_to_ann_tensor_size) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:65]: `current_size`?

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h [hunk 16]
```diff
@@ -72,15 +74,17 @@ class OperandMapping {
   // converted during copying the data to the memory allocated for NN API.
   // kTfLiteNoType means no conversion is needed.
   TfLiteType lite_index_to_ann_type_conversion(int index) const {
-    if (index >= 0 && index < index_to_type_conversion_.size())
+    const int64 index_to_type_conversion_size = index_to_type_conversion_.size();
+    if (index >= 0 && index < index_to_type_conversion_size)
```
Comment (mihaimaruseac) [inline tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:78]: `max_size`?

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h [hunk 17]
```diff
@@ -72,15 +74,17 @@ class OperandMapping {
...
       return index_to_type_conversion_[index];
     else
       return kTfLiteNoType;
   }
 
   // Add a new mapping from TFLite index to a type conversion.
   void add_type_conversion(int tflite_index, TfLiteType tflite_type) {
-    if (tflite_index >= index_to_type_conversion_.size()) {
+    const int64 index_to_type_conversion_size = index_to_type_conversion_.size();
+    if (tflite_index >= index_to_type_conversion_size) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:87]: `current_size`?

File: tensorflow/python/client/session_ref.cc (hunk 12 above)
Comment (mihaimaruseac) [reply tensorflow/python/client/session_ref.cc:150]: Keeping unsolved until that one lands

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/profiler/utils/event_span.cc (hunk 9 above)
Comment (tg-at-google) [reply tensorflow/core/profiler/utils/event_span.cc:132]: updated.

File: tensorflow/python/client/session_ref.cc (hunk 12 above)
Comment (tg-at-google) [reply tensorflow/python/client/session_ref.cc:150]: updated.

File: tensorflow/compiler/xla/index_util.cc (hunk 13 above)
Comment (tg-at-google) [reply tensorflow/compiler/xla/index_util.cc:162]: updated.

File: tensorflow/core/profiler/internal/tfprof_code.cc (hunk 14 above)
Comment (tg-at-google) [reply tensorflow/core/profiler/internal/tfprof_code.cc:432]: updated.

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h (hunk 15 above)
Comment (tg-at-google) [reply tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:65]: updated.

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h (hunk 16 above)
Comment (tg-at-google) [reply tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:78]: updated

File: tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h (hunk 17 above)
Comment (tg-at-google) [reply tensorflow/lite/delegates/nnapi/nnapi_delegate_kernel.h:87]: updated

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc (hunk 5 above)
Comment (mihaimaruseac) [reply tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1571]: Keeping open until rebase removes it from here

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
```
Executed 1217 out of 2545 tests: 1218 tests pass, 1292 fail to build and 35 fail remotely.
```

**Comment by mihaimaruseac** [comment]
Closing as it has been handled by new PRs.

**Pull request closed without merging**